test/integration/instance/save.test.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+var chai = require('chai')
+  , Sequelize = require('../../../index')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support')
+  , dialect = Support.getTestDialect()
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , Promise = Sequelize.Promise;
+
+chai.config.includeStack = true;
+
+describe(Support.getTestDialectTeaser('Instance'), function() {
+  describe('save', function() {
+    describe('include', function() {
+      it('should save data for BelongsTo relations', function() {
+        var Product = this.sequelize.define('Product', {
+          title: Sequelize.STRING
+        });
+        var User = this.sequelize.define('User', {
+          first_name: Sequelize.STRING,
+          last_name: Sequelize.STRING
+        });
+
+        Product.belongsTo(User);
+
+        var product = Product.build({
+          id: 1,
+          title: 'Chair',
+          User: {
+            id: 1,
+            first_name: 'Mick',
+            last_name: 'Hansen'
+          }
+        }, {
+          include: [ User ]
+        });
+
+        return this.sequelize.sync({ force: true }).then(function() {
+          return product.save().then(function(savedProduct) {
+            return Product.find({
+              where: { id: savedProduct.id },
+              include: [ User ]
+            }).then(function(persistedProduct) {
+              expect(persistedProduct.User).to.be.ok;
+            });
+          });
+        });
+      });
+
+      it('should save data for HasMany relations', function() {
+        var Product = this.sequelize.define('Product', {
+          title: Sequelize.STRING
+        });
+        var Tag = this.sequelize.define('Tag', {
+          name: Sequelize.STRING
+        });
+
+        Product.hasMany(Tag);
+
+        var product = Product.build({
+          id: 1,
+          title: 'Chair',
+          Tags: [
+            {id: 1, name: 'Alpha'},
```

Comment:
Does it work without the primary key?
If it does does it determine create/update based on the presense of foreign key alone and how we would work with that for non AI PKs like UUIDs?